tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -54,29 +58,96 @@ struct SynthOrdered {
     }
 };
 
-template <class T>
-inline constexpr bool is_pair = false;
-template <class A, class B>
-inline constexpr bool is_pair<std::pair<A, B>> = true; // TRANSITION, std::pair spaceship not yet implemented
+struct OrderedChar {
+    OrderedChar() = default;
+    OrderedChar(const char c) : c(c) {}
 
-template <class Container>
-void ordered_containers_test(const Container& smaller, const Container& smaller_equal, const Container& larger) {
+    OrderedChar& operator=(const char& other) {
+        c = other;
+        return *this;
+    }
+
+    auto operator<=>(const OrderedChar&) const = default;
+
+    operator char() const {
+        return c;
+    }
+
+    char c;
+};
+
+struct WeaklyOrderedChar : OrderedChar {};
+struct WeaklyOrderdByOmissionChar : OrderedChar {};
+struct PartiallyOrderedChar : OrderedChar {};
+
+namespace std {
+    template <>
+    struct char_traits<OrderedChar> : public char_traits<char> {
+        using char_type = OrderedChar;
+
+        static int compare(const char_type* first1, const char_type* first2, size_t count) {
+            for (; 0 < count; --count, ++first1, ++first2) {
+                if (*first1 != *first2) {
+                    return *first1 < *first2 ? -1 : +1;
+                }
+            }
+
+            return 0;
+        }
+
+        static bool eq(const char_type l, const char_type r) {
+            return l.c == r.c;
+        }
+    };
+
+    template <>
+    struct char_traits<WeaklyOrderedChar> : public char_traits<OrderedChar> {
+        using char_type           = WeaklyOrderedChar;
+        using comparison_category = weak_ordering;
+    };
+
+    template <>
+    struct char_traits<WeaklyOrderdByOmissionChar> : public char_traits<OrderedChar> {
+        using char_type = WeaklyOrderdByOmissionChar;
+    };
+
+    template <>
+    struct char_traits<PartiallyOrderedChar> : public char_traits<OrderedChar> {
+        using char_type           = PartiallyOrderedChar;
+        using comparison_category = partial_ordering;
+    };
+} // namespace std
+
+template <class ReturnType, class SmallType, class EqualType, class LargeType>
+void spaceship_test(const SmallType& smaller, const EqualType& smaller_equal, const LargeType& larger) {
+    assert(smaller == smaller_equal);
+    assert(smaller_equal == smaller);
+    assert(smaller != larger);
+    assert(larger != smaller);
     assert(smaller < larger);

```

Comment:
Now that we have cross-type comparisons, I think it would be best to add reversed comparisons for all operators. That is, testing `smaller != larger` and `larger != smaller` is great, so in addition to `smaller < larger`, we should test `!(larger < smaller)`. (I think you have 100% instantiation coverage due to all of the calls below, but adding reversed comparisons here is a simple way to be super duper sure.)